src/gui/transferlistsortmodel.cpp

```diff
@@ -166,8 +166,8 @@ bool TransferListSortModel::lessThan(const QModelIndex &left, const QModelIndex
 
             const qlonglong etaL = left.data().toLongLong();
             const qlonglong etaR = right.data().toLongLong();
-            const bool isInvalidL = ((etaL < 0) || (etaL >= MAX_ETA));
-            const bool isInvalidR = ((etaR < 0) || (etaR >= MAX_ETA));
+            const bool isInvalidL = ((etaL < 0) || (etaL >= MAX_ETA.count()));
+            const bool isInvalidR = ((etaR < 0) || (etaR >= MAX_ETA.count()));
```

Comment:
I'm suggesting you make `etaR` and `etaL` of type `std::duration`. That way, `operator>=` will do the right thing, regardless of the `MAX_ETA` unit.

right now, you rely on `MAX_ETA` specifically being `std::minutes`, if you change the value of this constant to be specified in seconds, it breaks.

Reply:
But then I have to ensure that when I pack the duration into `QVariant` and unpack here, I use the same type. Otherwise things break in a similar way. Or I can put a rule that `QVariant` always contains seconds, which doe not differ significantly from what I have now, except that I can check in an IDE that `MAX_ETA` is indeed seconds.

Reply:
I'm not familiar with `QVariant`, but right now it contains a raw integral, which the reader of this code needs to know the unit of. I'm suggesting that you "unpack" this integral into a chrono type. This will serve two purposes, it will increase the coverage of using proper types and improve type safety. It will also make it *explicit* that you are treating this raw integral as seconds or minutes (I forget which one it is), so it becomes easier for a reader to understand the intention or spot a bug if it's wrong.

fundamentally though, it would probably be better to store a `chrono::duration` type in the variant too

Reply:
you're right that as long as the variant is a raw integral, you can still make a mistake of treating it as the wrong unit, but as the code is now, you have more opportunities to make that mistake